centreon/www/include/configuration/configObject/host/DB-Func.php

```diff
@@ -497,6 +497,60 @@ function deleteHostInDB($hosts = []): void
     }
 }
 
+/**
+ * @param int[] $hosts
+ */
+function deleteHostInApi(array $hosts = []): void
+{
+    global $basePath;
+
+    try {
+        deleteHostByApi($basePath, $hosts);
+    } catch (Throwable $throwable) {
+        CentreonLog::create()->error(
+            logTypeId: CentreonLog::TYPE_BUSINESS_LOG,
+            message: "Error while deleting host by API : {$throwable->getMessage()}",
+            customContext: ['hosts' => $hosts, 'basePath' => $basePath],
+            exception: $throwable
+        );
+    }
+}
+
+/**
+ * @param string $basePath
+ * @param int[] $hosts
+ *
+ * @throws Throwable
+ */
+function deleteHostByApi(string $basePath, array $hosts): void
+{
+    $kernel = Kernel::createForWeb();
+    /** @var Router $router */
+    $router = $kernel->getContainer()->get(Router::class);
+    $client = new CurlHttpClient();
+
+    $headers = [
+        'Content-Type' => 'application/json',
+        'Cookie' => 'PHPSESSID=' . $_COOKIE['PHPSESSID'] . ';XDEBUG_SESSION=XDEBUG_KEY',
+    ];
+
+    foreach ($hosts as $hostId) {
+        $url = $router->generate(
+            'DeleteHost',
+            $basePath ? ['base_uri' => $basePath, 'hostId' => $hostId] : [],
+            UrlGeneratorInterface::ABSOLUTE_URL,
+        );
+
+        $response = $client->request('DELETE', $url, ['headers' => $headers]);
+
+        if ($response->getStatusCode() !== 204) {
+            $content = json_decode(json: $response->getContent(false), flags: JSON_THROW_ON_ERROR);
+
+            throw new Exception($content->message ?? 'Unexpected return status while deleting host');
```

Comment:
Are you sure you want to stop the deletion process in the event of a deletion error?
In any case, you can't do a rollback.